akka-actor/src/main/scala/akka/io/dns/DnsResourceRecords.scala

```diff
@@ -15,41 +15,46 @@ import akka.util.{ unused, ByteIterator, ByteString }
 import scala.annotation.switch
 import scala.concurrent.duration._
 
-sealed abstract class ResourceRecord(val name: String, val ttl: Ttl, val recType: Short, val recClass: Short)
+sealed abstract class ResourceRecord(
+    val name: String,
+    val ttl: CachePolicy.Ttl,
+    val recType: Short,
+    val recClass: Short)
     extends NoSerializationVerificationNeeded {}
 
-final case class ARecord(override val name: String, override val ttl: Ttl, ip: InetAddress)
+// TODO - should we stop these being case classes? And all other DNS classes?
```

Comment:
maybe on first modification as we already made the API stable

Reply:
yes, let's not break more than necessary